packages/@aws-cdk/aws-dynamodb/lib/table.ts

```diff
@@ -182,10 +182,245 @@ export interface LocalSecondaryIndexProps extends SecondaryIndexProps {
+   */
+  readonly tableName: string;
+
+  /**
+   * stream arn of the dynamodb table
+   *
+   * @attribute
+   */
+  readonly tableStreamArn?: string;
+
+  /**
+   * Adds an IAM policy statement associated with this table to an IAM
+   * principal's policy.
+   * @param grantee The principal (no-op if undefined)
+   * @param actions The set of actions to allow (i.e. "dynamodb:PutItem", "dynamodb:GetItem", ...)
+   */
+  grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant;
+
+  /**
+   * Adds an IAM policy statement associated with this table's stream to an
+   * IAM principal's policy.
+   * @param grantee The principal (no-op if undefined)
+   * @param actions The set of actions to allow (i.e. "dynamodb:DescribeStream", "dynamodb:GetRecords", ...)
+   */
+  grantStream(grantee: iam.IGrantable, ...actions: string[]): iam.Grant;
+
+  /**
+   * Permits an IAM principal all data read operations from this table:
+   * BatchGetItem, GetRecords, GetShardIterator, Query, GetItem, Scan.
+   * @param grantee The principal to grant access to
+   */
+  grantReadData(grantee: iam.IGrantable): iam.Grant;
+
+  /**
+   * Permits an IAM Principal to list streams attached to current dynamodb table.
+   *
+   * @param grantee The principal (no-op if undefined)
+   */
+  grantTableListStreams(grantee: iam.IGrantable): iam.Grant;
+
+  /**
+   * Permits an IAM principal all stream data read operations for this
+   * table's stream:
+   * DescribeStream, GetRecords, GetShardIterator, ListStreams.
+   * @param grantee The principal to grant access to
+   */
+  grantStreamRead(grantee: iam.IGrantable): iam.Grant;
+
+  /**
+   * Permits an IAM principal all data write operations to this table:
+   * BatchWriteItem, PutItem, UpdateItem, DeleteItem.
+   * @param grantee The principal to grant access to
+   */
+  grantWriteData(grantee: iam.IGrantable): iam.Grant;
+
+  /**
+   * Permits an IAM principal to all data read/write operations to this table.
+   * BatchGetItem, GetRecords, GetShardIterator, Query, GetItem, Scan,
+   * BatchWriteItem, PutItem, UpdateItem, DeleteItem
+   * @param grantee The principal to grant access to
+   */
+  grantReadWriteData(grantee: iam.IGrantable): iam.Grant;
+
+  /**
+   * Permits all DynamoDB operations ("dynamodb:*") to an IAM principal.
+   * @param grantee The principal to grant access to
+   */
+  grantFullAccess(grantee: iam.IGrantable): iam.Grant;
+}
+
+/**
+ * Reference to a dynamodb table.
+ */
+export interface TableAttributes {
+  /**
+   * The ARN of the dynamodb table.
+   *
+   * @default no table arn
+   */
+  readonly tableArn?: string;
+
+  /**
+   * The table name of the dynamodb table.
+   *
+   * @default no table name
+   */
+  readonly tableName?: string;
+
+  /**
+   * The stream arn of the dynamodb.
+   *
+   * @default no table stream arn
+   */
+  readonly tableStreamArn?: string;
+
+  /**
+   * Indicate if the table has global or local secondary indexes.
+   */
+  readonly hasIndex: boolean;
```

Comment:
I understand the intention here, but I don't think this is the way to go here. I think what you want here instead is:


```ts
/** @default none */
readonly localSecondaryIndexes?: string[];

/** @default none */
readonly globalSecondaryIndexes?: string[];
```

*However*, I don't think we need these in the initial version. We can always add them later.

Reply:
The shared `public grant(grantee: iam.IGrantable, ...actions: string[]): iam.Grant` method depends on the flag if the table has indexes to grant the permissions to `IGrantable`.

Indeed we can have both `localSecondaryIndexes`  and `globalSecondaryIndexes` in given `TableAttrs`, I think the attrs of imported dynamodb would be more complex and redundancy.

Reply:
Right. And for imported Tables, the check will be the same: `localSecondaryIndexes.length + globalSecondaryIndexes.length > 0`. The indexes are needed for things like auto-scaling policies.

But like I said - this isn't really required for the first version, feel free to leave it out (and just grant permissions without indexes for imported Tables).